client/src/components/headerComponents/Header.js

```diff
@@ -0,0 +1,67 @@
+import './Header.css';
+import React from "react";
+import {
+	IconButton,
+	HStack,
+	useColorMode,
+	InputGroup,
+	Input,
+	InputLeftElement,
+	VStack,
+	Button
+} from "@chakra-ui/react";
+import { FaSun, FaMoon, FaSearch } from "react-icons/fa";
+import { Formik, Form, Field } from 'formik';
+import * as Yup from 'yup';
+import { v4 as uuidv4 } from 'uuid';
+
+const videObjSchema = Yup.object().shape({
+	id: Yup.number().required('Required'),
+	time: Yup.date(),
+	title: Yup.string().min(3, 'Too short!').max(100, 'Too long!').required('Required'),
+	url: Yup.string().max(100).url().required('Please enter valid URL'),
+	rating: Yup.number().required('Required')
+})
+
+const Header = ({ setVideosArr, videosArr }) => {
+	const { colorMode, toggleColorMode } = useColorMode();
+
+	const ratingNumber = () => Number(Math.floor(Math.random() * 10000));
+
+	return (
+		<VStack>
+			<HStack p='1rem' w='100%' display='flex' justifyContent='space-between'>
+				<InputGroup w='60vw'>
+					<InputLeftElement pointerEvents='none' children={<FaSearch />} />
+					<Input />
+				</InputGroup>
+				<IconButton icon={colorMode === 'light' ? <FaSun /> : <FaMoon />} alignSelf='flex-end' onClick={toggleColorMode} isRound='true' />
+			</HStack>
+			<Formik initialValues={{
+				id: 0,
+				title: '',
+				url: '',
+				rating: 0
+			}} validationSchema={videObjSchema} onSubmit={values => {
```

Comment:
I have checked this package *yup. That's pretty interesting in the meantime, It's a bit confusing. when you have time you may show me how it works that would be great.